libcontainer/configs/cgroup.go

```diff
@@ -57,6 +57,11 @@ type Cgroup struct {
 	// MEM to use
 	CpusetMems string `json:"cpuset_mems"`
 
+	// Process limit; set to `0' to disable limit.
+	// While technically `0' is a valid PID limit, it does not make sense in the
+	// context of a container -- it is identical to a limit of `1'.
```

Comment:
@hqhq ```pids.max = 0``` does not means there is no process in a cgroup or container. It does not work on attached process or a new attaching operation, pids.current probably larger than pids.max.

Let me try to explain more about what I suggest for container here. 
```
pids.Limit = N (N > 0) means there are at most N pids in this container, only checking fork() or clone().
pids.Limit = 0 means No fork() or clone() is allowed.
pids.Limit < 0 means No limit in pids
```
@cyphar As you said, "Having a limit of 0 is precisely identical to having a limit of 1. Both prevent fork() and clone() in a container with a single process in it.". Yes, we get the same result, but we did different things. One is to prevent fork() and clone(), the other one is to allow fork() and clone() but the limit is always reached when you want to do it.

Okey let's see the cgroup again.
```
pids.max = N (N > 0) means there are at most N pids in this group, only check in copy_process().
pids.max = 0 means No more copy_process() is allowed here.
pids.max = "max" means No pids limit at all.
```
Let's think about the ```pids.max = 0``` and ```pids.max = 1```, we can get the same conclusion with container case, they are working equally. When there is no process in cgroup, both of them would work, when there is any process in cgroup, both of them are preventing user to fork() or clone(). Buuuuuuuuuuut, Why TJ suggested "max" rather than 0? Because pids.max = 0 has a special meaning. 

So, 
(1). Yes, we can get what we want with pids.max = 1, but they are different in semasiology.
(2). I think it's better to keep the behaviour in container with what it is in cgroup.

BTW, @cyphar why did you only check pids limit in copy_process(), ignoring to check attaching new pids and setting pids.max? This design would cause pids.current > pids.max which is really confusing.

Thanx
Yang